### ISSUE TITLE
XML Documentation missing for packaged FSharp.Core

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk#13590]
The F# Compiler and F# Interactive Repl bundle a copy of FSharp.Core with them, which is used as the 'canonical' FSharp.Core reference for F# Interactive.

Tools like Ionide for VS Code, which use the F# Autocomplete library to provide intellisense for F# Scripts, use this same FSharp.Core dll in order to provide an autocomplete experience that exactly matches the one the user would get if they were using F# Interactive directly when authoring scripts.

Right now, users cannot get documentation tooltips for scripts because this packaging of FSharp.Core doesn't include the XML documentation files that are packaged in the consumed [Microsoft.FSharp.Compiler](https://github.com/dotnet/fsharp/blob/main/src/fsharp/FSharp.Compiler.nuget/Microsoft.FSharp.Compiler.nuspec#L52) nuget package by the [tool_fsc](https://github.com/dotnet/sdk/blob/7ee5193dc4f6b19db35f886e6dc6d65bbccf3589/src/Layout/tool_fsharp/tool_fsc.csproj#L7) project.

This project is built and published [here](https://github.com/dotnet/sdk/blob/ad022a2912fb95c506787e2c6a5bcc0d9aeccaff/src/Layout/redist/targets/GenerateLayout.targets#L114-L120), but publishing doesn't include the xml documentation (or PDBs) for consumed libraries by default.

I'd like to request that the XML documentation file for FSharp.Core be added to this packaging, so that users of Ionide for VSCode, Emacs, Vim, and other editors can have a consistent autocomplete experience no matter what tooling they use.

In my estimation it _could_ be as simple as a snippet like this added to the project file:

```xml
<Target Name="_ResolvePublishNuGetPackageXml"
        AfterTargets="RunResolvePublishAssemblies">
  <ItemGroup>
      <ResolvedFileToPublish
        Include="@(ResolvedAssembliesToPublish->'%(RootDir)%(Directory)%(Filename).xml')"
        RelativePath="$([System.IO.Path]::ChangeExtension(%(ResolvedAssembliesToPublish.DestinationSubPath), '.xml'))"
        DestinationSubPath="$([System.IO.Path]::ChangeExtension(%(ResolvedAssembliesToPublish.DestinationSubPath), '.xml'))"
        Condition="'%(ResolvedAssembliesToPublish.PackageName)' != ''
                    and Exists('%(RootDir)%(Directory)%(Filename).xml')" />
  </ItemGroup>
</Target>
```

but I'm sure there's a better way to do things.

**Comment by wli3** [comment]
cc @brettfo @KevinRansom

**Comment by brettfo** [comment]
This was fixed in dotnet/fsharp#10121 and got merged to our 16.8 branch in dotnet/fsharp#10124.  That PR corresponds to [this build](https://dev.azure.com/dnceng/internal/_build/results?buildId=815436&view=results) and is in package version `Microsoft.FSharp.Compiler.11.0.0-beta.20464.8`.

Not sure when/if that's been merged into this repo yet, so I'll pass back to @wli3.

**Comment by baronfel** [comment]
This repo has been updated past that version (the current version is 11.0.0-beta.20465.1), but as I mentioned the crux of the problem is that `dotnet publish` doesn't include xml docs for referenced libraries.

I cloned this repo, and ran `dotnet publish -c Release` in the `src/Layout/tool_fsharp` directory to replicate what happens in the layout targets. The project outputs published artifacts at `<root>/artifacts/bin/tool_fsc_/<configuration>/<tfm>/publish`, so we can see if the xmldocs are present:

```
> ls /Users/chethusk/oss/dotnet-sdk/artifacts/bin/tool_fsc/Release/net5.0/publish/FSharp.Core.*
/Users/chethusk/oss/dotnet-sdk/artifacts/bin/tool_fsc/Release/net5.0/publish/FSharp.Core.dll
```

They are not.

So then I checked the referenced nuget package to verify that the docs were there:

```
> ls ~/.nuget/packages/microsoft.fsharp.compiler/11.0.0-beta.20465.1/lib/netcoreapp3.1/FSharp.Core.*
/Users/chethusk/.nuget/packages/microsoft.fsharp.compiler/11.0.0-beta.20465.1/lib/netcoreapp3.1/FSharp.Core.dll
/Users/chethusk/.nuget/packages/microsoft.fsharp.compiler/11.0.0-beta.20465.1/lib/netcoreapp3.1/FSharp.Core.xml
``` 

They are.

So in the end I needed to add a target that would add that xml doc to the set of files to be published (based on a slightly derivative version of the comment linked in my original submission) to get the xml doc to be published. I'll submit a PR for review shortly.
